test/test_optim.py

```diff
@@ -2163,6 +2165,260 @@ def test_non_empty_state(self, device, dtype, optim_info):
             for state in optim.state.values():
                 self.assertGreater(len(state), 0)
 
+    @optims(
+        [optim for optim in optim_db if optim.optim_cls.__name__ == "Adam"],
+        dtypes=[torch.float32],
+    )
+    def test_less_mem_beta1_zero_adam(self, device, dtype, optim_info):
+        # test that beta1=0.0 uses less memory than the default
+        model = torch.nn.Linear(5, 5)
+        model.to(dtype=dtype, device=device)
+        inpt = torch.rand(2, 5, dtype=dtype, device=device)
+
+        optim_inputs = optim_info.optim_inputs_func(device=device)
+        case_to_mem_usage = {}
+        for optim_input in optim_inputs:
+            beta1_zero = (
+                "betas" in optim_input.kwargs and optim_input.kwargs["betas"][0] == 0.0
+            )
+            if beta1_zero or optim_input.desc == "default":
+                activities = (
+                    [ProfilerActivity.CUDA]
+                    if device == "cuda"
+                    else [ProfilerActivity.CPU]
+                )
+
+                with profile(
+                    activities=activities, profile_memory=True, record_shapes=True
+                ) as prof:
+                    optim = optim_info.optim_cls(
+                        model.parameters(), **optim_input.kwargs
+                    )
+                    optim.zero_grad()
+                    output = model(inpt)
+                    loss = output.sum()
+                    loss.backward()
+
+                    optim.step()
+
+                case_to_mem_usage["beta1-zero" if beta1_zero else "default"] = sum(
+                    [
+                        item.cuda_memory_usage
+                        if device == "cuda"
+                        else item.cpu_memory_usage
+                        for item in prof.key_averages()
+                    ]
+                )
+
+        self.assertGreater(
```

Comment:
Could we get more precise about memory here? We should be able to figure out the size of exp_avg precisely + test against that.

Reply:
Thanks for your reply. I made the change in my last commit. Let me know if further adjustments are needed.